plugins/config_manager.py

```diff
@@ -78,6 +78,31 @@ def run(self):
     def modify_repo(self):
         """ process --set-enabled, --set-disabled and --setopt options """
 
+        matching_repos = []         # list of matched repositories
+        not_matching_repos_id = set()  # IDs of not matched repositories
+
+        def match_repos(key, add_matching_repos):
+            matching = self.base.repos.get_matching(key)
+            if not matching:
+                not_matching_repos_id.add(name)
+            elif add_matching_repos:
+                matching_repos.extend(matching)
+
+        if self.opts.crepo:
```

Comment:
I'm thinking that if we extended `matching_repos` with repoids from `repo_setopts` unconditionally (that is, regardless of whether `self.opts.crepo` is empty or not), we could

* simplify the code quite a bit (the `else` clause could be removed and `match_repos()` could be simplified if not removed as well) and
* make the behavior compatible with the original `yum-config-manager` in RHEL-7.

The difference currently is in the way we treat these repoids: here, we use them to **initialize the default** argument list, whereas in RHEL-7, we use them simply to **extend** the final argument list. The rationale behind the latter is that once you use a repoid-flavoured `--setopt`, you expect that to be applied, no matter what other repos you list as arguments.

In the nutshell, `--setopt=c.gpgcheck=0 a b` would currently not alter `c`. This limitation is not documented in the man page (as updated in this PR). As an alternative to what I proposed above, we could just add a note note to the `--setopt` section about this limitation, but I'm afraid it would just make it all more confusing to the reader.